packages/material-ui/src/IconButton/IconButton.js

```diff
@@ -87,6 +93,7 @@ function IconButton(props) {
         {
           [classes[`color${capitalize(color)}`]]: color !== 'default',
           [classes.disabled]: disabled,
+          [classes.sizeSmall]: size === 'small',
```

Comment:
What do you think of 
```suggestion
          `size${capitalize(size)}`
```
and add `sizeMedium`. Makes it more obvious what one has to override if the size is `medium`. Then add a test for no size and `size="medium".`

Reply:
@eps1lon If we want to be consistent with the other components, we shouldn't add a new class key for the default value. it's making override harder.

Reply:
... 24 seconds after I added this :smile:

Reply:
Overriding is as simple as before, because `sizeMedium` does nothing by default.

Reply:
I can provide an example regarding the `theme.overrides` problem.

Reply:
It's a good occasion to pick a pattern, once for all.

Reply:
Just saw that `Button` only has sizeXY for XY !== medium. I'll update IconButton then.

Reply:
Feel free to revert the last commit before merging this, I don't know what's the best solution here and why.

Reply:
It's still not addressing that it's not obvious what class I have to override if `size="medium"`.

Reply:
That's an issue with all components that don't provide a class for the default case. What is the problem with `theme.overrides` if we add that classes? @oliviertassinari

Reply:
Let's take the Button component as a baseline. Here what we do, the default medium size doesn't have any specific class, the style is directly hosted in `root`. Let's consider the use cases now.
- Case n°1: I want to override the button padding with the theme. I have to do:
```js
const theme = createMuiTheme({
  overrides: {
    MuiButton: {
      root: {
        padding: 8,
      },
    },
  },
});
```
- Case n°2: I want to override the button padding with `withStyles()`. I have to do:
```js
const Button = withStyles({
  root: {
    padding: 8,
  },
})(MuiButton)
```

In both cases, I can completely ignore what the default properties are. Now, let's say I want to customize the different size values: I would need to do:
```jsx
root: {
  padding: 8,
},
sizeSmall: {
  padding: 4,
},
sizeLarge: {
  padding: 16,
},
```

or
```js
const sizes = {
  small: 4,
  medium: 8,
  large: 16,
}

root: {
  padding: props => sizes[props.size],
}
``` 

Doing the following won't work because we don't have a `sizeMedium` key:
```js
sizeSmall: {
  padding: 4,
},
sizeMedium: {
  padding: 8,
},
sizeLarge: {
  padding: 16,
},
```
I guess we could add it for consistency, why not.

Ok now, what's the problem with writing the styles like this in the core implementation?:
```js
sizeSmall: {
  padding: 4,
},
sizeMedium: {
  padding: 8,
},
sizeLarge: {
  padding: 16,
},
```
We break case n°1. The correct solution become:
```js
const theme = createMuiTheme({
  overrides: {
    MuiButton: {
      sizeMedium: {
        padding: 8,
      },
    },
  },
});
```
While case n°2 solution is still valide.

Reply:
Agree with point one but why not follow through and put the style into the responsible class? The default style should be handled by what's defined in `defaultProps`. I don't understand why we add special treatment for the default case. Just put the styles into the responsible class and let the props handle what class is applied.

It's otherwise not obvious what is responsible for applying the style for the default size. You would look into `sizeMedium` and find nothing there. `sizeSmall` has `padding: 8`. I would say intuitively the actual style for `size="medium"` should have no padding. Makes it unnecessarily hard to quickly scan what what `size` applies which padding.

Reply:
> I don't understand why we add special treatment for the default case.

So people can override as much style as possible from one class, not multiples, exactly like Bootstrap.

> It's otherwise not obvious what is responsible for applying the style for the default size. You would look into `sizeMedium` and find nothing there. `sizeSmall` has `padding: 8`. I would say intuitively the actual style for `size="medium"` should have no padding. Makes it unnecessarily hard to quickly scan what what `size` applies which padding.

That's optimizing for people who read the source, I wouldn't consider this dimension the most important. I would rather optimize for people using the dev tools.

Reply:
> That's optimizing for people who read the source, I wouldn't consider this dimension the most important. I would rather optimize for people using the dev tools.

I wasn't thinking about this case. I can read the classnames and compare with our docs. Don't have to read source.

Reply:
> I can read the classnames from browser devtools and compare with our docs.

This is how I style components, at least.

Reply:
I revert what I said regarding the style = f(props) part, it's not valid. The only valid part is with `theme.overrides` usage. No matter what we pick, we need to be consistent, it's probably more important than the actual solution we go for. Ok, I'm happy to go with the most intuitive option, I believe it's what the majority of our components are doing. While we are at it. The Paper square case is interesting to consider. How would you handle it?